File: 7kyu/fix-string-case/fix-string-case.py
import re
def solve(s):
    lower = re.compile(r'[a-z]')
    lower_count = len(re.findall(lower, s))
    upper_count = len(s) - lower_count
    return s.lower() if lower_count >= upper_count else s.upper()

# print(solve("code"))
print(solve("code") == "code")
print(solve("CODe") == "CODE")
print(solve("COde") == "code")

File: 7kyu/fix-string-case/fix-string-case.go
package main

import (
	"fmt"
	"regexp"
	"strings"
)

func solve(str string) string {
	lowerRegex := regexp.MustCompile(`[a-z]`)
	lowerCount := len(lowerRegex.FindAllString(str, -1))
	upperCount := len(str) - lowerCount

	if lowerCount >= upperCount {
		return strings.ToLower(str)
	} else {
		return strings.ToUpper(str)
	}
}

func main() {
	fmt.Println(solve("CODe"))
}
